Noura_Alhadlaq/week_01/day_04/Homework4/main.js

```diff
@@ -0,0 +1,107 @@
+
+//Part 1: Array Algorithms
+
+function sumArray(array) {
+    var result = 0;
+    for (var i = 0; i < array.length; i++) {
+        result += array[i];
+    } console.log(result);
+    return result;
+}
+sumArray([1, 2, 3, 4]);
+
+
+function multiplyArray(array) {
+    var result = 1;
+    for (var i = 0; i < array.length; i++) {
+        result = result * array[i];
+    } console.log(result);
+    return result;
+}
+
+multiplyArray([1, 2, 3, 4]);
+
+// Write a function findLongestWord that takes an array of words and 
+// returns the length of the longest word in the array.
+
+function findLongestWord(array) {
+    var count = 0;
+    var word;
+    for (i = 0; i < array.length; i++) {
+        if (array[i].length > count) {
+            count += array[i].length;
```

Comment:
Almost got the perfect solution here, re-evaluate this line `count += array[i].length;`, because if I call `findLongestWord(['a', 'bb', 'ccc']);` it'll give me: `bb` where it should've been `ccc`. How can we solve this?

Reply:
count = array[i].length;